cpp/simulations/abm.cpp

```diff
@@ -0,0 +1,562 @@
+        person.set_assigned_location(event);
+        person.set_assigned_location(shop);
+        counter_shop++;
+        //assign hospital and ICU
+        person.set_assigned_location(hospital);
+        person.set_assigned_location(icu);
+        //assign work/school to people depending on their age
+        if (person.get_age() == mio::abm::AgeGroup::Age5to14) {
+            person.set_assigned_location(school);
+            counter_school++;
+        }
+        if (person.get_age() == mio::abm::AgeGroup::Age15to34 || person.get_age() == mio::abm::AgeGroup::Age35to59) {
+            person.set_assigned_location(work);
+            counter_work++;
+        }
+        //add new school/work/shop if needed
+        if (counter_school == 600) {
+            counter_school = 0;
+            school         = world.add_location(mio::abm::LocationType::School);
+            world.get_individualized_location(school).get_infection_parameters().set<mio::abm::MaximumContacts>(40);
+        }
+        if (counter_work == 100) {
+            counter_work = 0;
+            work         = world.add_location(mio::abm::LocationType::Work);
+            world.get_individualized_location(work).get_infection_parameters().set<mio::abm::MaximumContacts>(40);
+        }
+        if (counter_shop == 15000) {
+            counter_shop = 0;
+            shop         = world.add_location(mio::abm::LocationType::BasicsShop);
+            world.get_individualized_location(shop).get_infection_parameters().set<mio::abm::MaximumContacts>(20);
+        }
+    }
+
+    // add the testing schemes for school and work
+    auto test_at_school = std::vector<mio::abm::LocationType>{mio::abm::LocationType::School};
+    auto testing_criteria_school =
+        std::vector<mio::abm::TestingCriteria>{mio::abm::TestingCriteria({}, test_at_school, {})};
+
+    testing_min_time           = mio::abm::days(7);
+    probability                = 1;
+    auto testing_scheme_school = mio::abm::TestingScheme(testing_criteria_school, testing_min_time, start_date,
+                                                         end_date, test_type, probability);
+    world.get_testing_strategy().add_testing_scheme(testing_scheme_school);
+
+    auto test_at_work = std::vector<mio::abm::LocationType>{mio::abm::LocationType::Work};
+    auto testing_criteria_work =
+        std::vector<mio::abm::TestingCriteria>{mio::abm::TestingCriteria({}, test_at_work, {})};
+
+    testing_min_time = mio::abm::days(1);
+    probability      = 0.5;
+    auto testing_scheme_work =
+        mio::abm::TestingScheme(testing_criteria_work, testing_min_time, start_date, end_date, test_type, probability);
+    world.get_testing_strategy().add_testing_scheme(testing_scheme_work);
+}
+
+/**
+ * Assign an infection state to each person.
+ */
+
+void assign_infection_state(mio::abm::World& world, double exposed_pct, double infected_pct, double carrier_pct,
+                            double recovered_pct)
```

Comment:
Here we also lack new additions from main. Please do a manual compare with the latest example.